packages/webapp/src/_app/eos/sessions.ts

```diff
@@ -194,7 +223,152 @@ const serializeSignatureAuth = async (
     return buffer.asUint8Array();
 };
 
-const signSha = (sessionKey: SessionKeyData, sha: number[]) => {
+const signData = async (
+    sessionKey: SessionKeyData,
+    data: Uint8Array
+): Promise<string> => {
     const privateKey = PrivateKey.fromString(sessionKey.privateKey);
-    return privateKey.sign(sha, false);
+    const signature = privateKey.sign(data, true);
+    console.info("eos signature >>>", signature, signature.toString());
+
+    if (sessionKey.subtleKey?.privateKey) {
+        console.info("i will sign with subtle >>>", sessionKey.subtleKey);
+
+        const privateKey = sessionKey.subtleKey.privateKey;
+        const signature = await crypto.subtle.sign(
+            { name: "ECDSA", hash: "SHA-256" },
+            privateKey,
+            data
+        );
+        const signatureBytes = new Uint8Array(signature);
+
+        const eosPubKey = await subtleToEosPublicKey(
+            sessionKey.subtleKey.publicKey!
+        );
+
+        // const pubKey = EC.keyFromPublic(
+        //     eosPubKey.data.subarray(0, 33)
+        // ).getPublic();
+        // const hash = new Uint8Array(
+        //     await crypto.subtle.digest("SHA-256", data)
+        // );
+
+        // const ecSignature = {
+        //     r: signatureBytes.slice(0, 32),
+        //     s: signatureBytes.slice(32, 64),
+        // };
+        // console.info(ecSignature, eosPubKey, pubKey, hash, signatureBytes);
+        // const recid = EC.getKeyRecoveryParam(hash, ecSignature, pubKey);
+        // const recidByte = 27 + 4 + recid;  <<<--- still presenting failures
+
+        let recId = 31;
+
+        const sigDataWithRecovery = {
+            type: KeyType.r1,
+            data: new Uint8Array([recId].concat(Array.from(signatureBytes))),
+        };
+        console.info(
+            "signature from subtle",
+            recId,
+            signatureBytes,
+            signature,
+            sigDataWithRecovery,
+            signatureToString(sigDataWithRecovery)
+        );
+
+        const sessionPubKeyString = publicKeyToString(eosPubKey);
+        let signatureStr = signatureToString(sigDataWithRecovery);
+        let signatureEcObj = Signature.fromString(signatureStr);
+        let recoveredKey = signatureEcObj.recover(data, true).toString();
+
+        console.info(
+            "recoveredKey >>>",
+            recoveredKey.toString(),
+            sessionPubKeyString
+        );
+
+        if (recoveredKey.toString() === sessionPubKeyString) {
+            return signatureStr;
```

Comment:
still gives me the same error above, even when I can confirm the recovered public key from elliptic lib matches.